utils/files/files.go

```diff
@@ -20,29 +20,34 @@ import (
 	"os"
 	"path/filepath"
 
-	"github.com/aws/aws-sdk-go/service/s3/s3manager"
 	"github.com/sirupsen/logrus"
 	"github.com/spf13/viper"
 )
 
-// This function takes a key from s3 and prepares the system for
-// download by recursively creating the path needed.
-func FilePathSetup(key *string, dwn *s3manager.Downloader) (*os.File, string) {
-
-	download_dir := viper.GetString("download_dir")
-	err := os.MkdirAll(download_dir+filepath.Dir(*key), 0700)
-	if err != nil {
-		logrus.Fatal("An error ocurred creating paths", err)
+// Creates a folder/file given the path in a recursive way
+// If the path finish with '/' it will create a file
+// Otherwise it will create a folder
+func FilePathSetup(absoluteFilePath string) *os.File {
+	if err := os.MkdirAll(filepath.Dir(absoluteFilePath), 0750); err != nil {
+		if err != nil {
```

Comment:
This is checking twice whether err is not nil

Reply:
Oh yeah, you're right. I didn't realize. Changed. Thank you.